Fix spurious 'Unrecognized config option' warning for wave.build.compression and wave.build.conda (#7355)
nextflow-io/nextflow: docs/developer/config-scopes.mdx

See `ExecutorConfig` and `ExecutorRetryConfig` for an example of how a nested scope is defined and constructed.

### Nested scopes from external types

> Sometimes a nested scope is backed by a type that cannot implement `ConfigScope`, such as a class from an external library. In this case, the field can be annotated with `@NestedScope` to mark it as a nested scope without requiring its type to implement `ConfigScope`.

Reviewer: ```suggestion
Some nested scopes are backed by a type that cannot implement `ConfigScope`, such as a class from an external library. Annotate the field with `@NestedScope` to mark it as a nested scope without requiring its type to implement `ConfigScope`.
```

> Each non-static instance field of the annotated field's type becomes a config option of the resulting scope. The field name determines the name of the nested scope, and the description (if any) is taken from the `@Description` annotation on the field.

Reviewer: ```suggestion
Each non-static instance field of the annotated field's type becomes a config option of the resulting scope. The field name determines the name of the nested scope, and the description (if any) comes from the `@Description` annotation on the field.
```

> For example, in the `wave` scope the `build.conda` and `build.compression` options are backed by wave-api types (`CondaOpts` and `BuildCompression`) that live in an external library:

Reviewer: ```suggestion
For example, in the `wave` scope, the `build.conda` and `build.compression` options are backed by wave-api types (`CondaOpts` and `BuildCompression`) defined in an external library:
```

```groovy
    @NestedScope
    @Description('''
        The `wave.build.compression` scope controls build image compression, e.g. `wave.build.compression.mode = 'estargz'`.
    ''')
    final BuildCompression compression
```

This exposes `wave.build.compression.mode`, `wave.build.compression.level`, and `wave.build.compression.force` as recognized config options.

> Because the config options are derived from the external type's fields, inner field descriptions and inner nested scopes are not supported with `@NestedScope`. Prefer a dedicated `ConfigScope` class where possible.

Reviewer: ```suggestion
This exposes `wave.build.compression.mode`, `wave.build.compression.level`, and `wave.build.compression.force` as recognized config options.

Because config options are derived from the external type's fields, `@NestedScope` does not support inner field descriptions or inner nested scopes. Prefer a dedicated `ConfigScope` class where possible.
```
